StRoot/StEvent/StFcsHit.h

```diff
@@ -87,9 +87,13 @@ class StFcsHit : public StObject {
     void setCluster(StFcsCluster* clu) {mCluster = clu;}
     StFcsCluster *cluster() {return mCluster;}
 
+    const vector<pair<unsigned int, float>>& getGeantTracks() const {return mGeantTracks;}
```

Comment:
Yes, it is needed.  A single ID truth is insufficient to understand the relative contributions of electromagnetic and hadronic components to the cluster energy response  (You don't know what the "rest" of the energy is...)   

With the list of track IDs hitting each cluster, their contribution to the energy deposition in the cluster, and the MC track's kinematics from the MC event record, you have the complete information needed to understand the clustering performance.

Reply:
> Yes, it is needed. A single ID truth is insufficient to understand the relative contributions of electromagnetic and hadronic components to the cluster energy response (You don't know what the "rest" of the energy is...)

Okay, but if that is the case, I need to also save PDG id. If not, I need to look into geant.root, and at this point I might as well not save anything to event.root. Right?

> With the list of track IDs hitting each cluster, their contribution to the energy deposition in the cluster, and the MC track's kinematics from the MC event record, you have the complete information needed to understand the clustering performance.


Again, for any foreseeable study, there are just two particles that could merge into the same cluster, one fraction per cell describes that case it well enough.

Reply:
> Okay, but if that is the case, I need to also save PDG id. If not, I need to look into geant.root, and at this point I might as well not save anything to event.root. Right?

PDG ID is available in the MuDst MuMcTrack branch.

> Again, for any foreseeable study, there are just two particles that could merge into the same cluster, one fraction per cell describes that case it well enough.

Relative contribution of hadronic and electromagnetic particles to the energy of a cluster is one such study which requires this information.

Reply:
> PDG ID is available in the MuDst MuMcTrack branch.

I'm not very familiar with that facility. And this answer also implies that this structure will be propagated to MuDst.

> 
> > Again, for any foreseeable study, there are just two particles that could merge into the same cluster, one fraction per cell describes that case it well enough.
> 
> Relative contribution of hadronic and electromagnetic particles to the energy of a cluster is one such study which requires this information.

Okay, so that would be a study that looks at cluster merging and hadronic/electromagnetic fractions at the same time. Thank you for the explanations. I hope other experts/analyzers can chime in.

Reply:
- Yes, idTruth is not enough, especially when many towers contributing to a cluster, and hadron showers are fat and big.
  - Yes, when we get to mudst, since Mudst has StMcTrack, but not g2t_hits.
   - No, when reading event.root file and geant.root file in sync. But this means we essentially re-run StFcsFastSimulationMake and creating new StFcsHits. (for historical reason, we don't have StMcTrack in StEvent).